metadata/src/from/v14.rs

```diff
@@ -0,0 +1,318 @@
+        v14::StorageEntryType::Plain(ty) => StorageEntryType::Plain(ty.id),
+        v14::StorageEntryType::Map {
+            hashers,
+            key,
+            value,
+        } => StorageEntryType::Map {
+            hashers: hashers.into_iter().map(from_storage_hasher).collect(),
+            key_ty: key.id,
+            value_ty: value.id,
+        },
+    }
+}
+
+fn from_storage_entry_modifier(value: v14::StorageEntryModifier) -> StorageEntryModifier {
+    match value {
+        v14::StorageEntryModifier::Optional => StorageEntryModifier::Optional,
+        v14::StorageEntryModifier::Default => StorageEntryModifier::Default,
+    }
+}
+
+fn from_storage_entry_metadata(
+    name: ArcStr,
+    s: v14::StorageEntryMetadata<PortableForm>,
+) -> StorageEntryMetadata {
+    StorageEntryMetadata {
+        name,
+        modifier: from_storage_entry_modifier(s.modifier),
+        entry_type: from_storage_entry_type(s.ty),
+        default: s.default,
+        docs: s.docs,
+    }
+}
+
+fn from_constant_metadata(
+    name: ArcStr,
+    s: v14::PalletConstantMetadata<PortableForm>,
+) -> ConstantMetadata {
+    ConstantMetadata {
+        name,
+        ty: s.ty.id,
+        value: s.value,
+        docs: s.docs,
+    }
+}
+
+fn generate_outer_enums(
+    metadata: &mut v14::RuntimeMetadataV14,
+) -> Result<frame_metadata::v15::OuterEnums<scale_info::form::PortableForm>, TryFromError> {
+    let outer_enums = crate::utilities::OuterEnums::find_in(&metadata.types);
+
+    let Some(call_enum_id) = outer_enums.call_ty else {
+        return Err(TryFromError::TypeNameNotFound("RuntimeCall".into()));
+    };
+    let Some(event_type_id) = outer_enums.event_ty else {
+        return Err(TryFromError::TypeNameNotFound("RuntimeEvent".into()));
+    };
+    let error_type_id = if let Some(id) = outer_enums.error_ty {
+        id
+    } else {
+        let call_enum = &metadata.types.types[call_enum_id as usize];
+        let mut error_path = call_enum.ty.path.segments.clone();
+
+        let Some(last) = error_path.last_mut() else {
+            return Err(TryFromError::InvalidTypePath("RuntimeCall".into()));
+        };
+        "RuntimeError".clone_into(last);
+        generate_outer_error_enum_type(metadata, error_path)
+    };
+
+    Ok(frame_metadata::v15::OuterEnums {
+        call_enum_ty: call_enum_id.into(),
+        event_enum_ty: event_type_id.into(),
+        error_enum_ty: error_type_id.into(),
+    })
+}
+
+/// Generates an outer `RuntimeError` enum type and adds it to the metadata.
+///
+/// Returns the id of the generated type from the registry.
+fn generate_outer_error_enum_type(
+    metadata: &mut v14::RuntimeMetadataV14,
+    path_segments: Vec<String>,
+) -> u32 {
+    let variants: Vec<_> = metadata
+        .pallets
+        .iter()
+        .filter_map(|pallet| {
+            let error = pallet.error.as_ref()?;
+
+            // Note:  using the `alloc::format!` macro like in `let path = format!("{}Error", pallet.name);`
```

Comment:
hmm interesting

Reply:
I wonder if that's still true; it's something Tadeo/Pavlo ran into but was very strange. Let me see if simplifying works